libmamba/src/core/subdirdata.cpp

```diff
@@ -10,103 +10,154 @@
 #include "mamba/core/output.hpp"
 #include "mamba/core/package_cache.hpp"
 #include "mamba/core/subdirdata.hpp"
+#include "mamba/core/thread_utils.hpp"
 #include "mamba/util/string.hpp"
 #include "mamba/util/url_manip.hpp"
 
-#include "progress_bar_impl.hpp"
-
 namespace mamba
 {
-    void subdir_metadata::serialize_to_stream(std::ostream& out) const
+    /*******************
+     * MSubdirMetadata *
+     *******************/
+
+    namespace
     {
-        nlohmann::json j;
-        j["url"] = url;
-        j["etag"] = etag;
-        j["mod"] = mod;
-        j["cache_control"] = cache_control;
-        j["size"] = stored_file_size;
+#ifdef _WIN32
+        std::chrono::system_clock::time_point filetime_to_unix(const fs::file_time_type& filetime)
```

Comment:
Put and test in the `util` library?

Reply:
This function is really local to this file, and was taken as is from the previous version. I think that on the long run, we want to get rid of it, so we may not expose it now.